### PR TITLE
Adding get_interfaces_ip and tests

### PR DESCRIPTION
**Pull request by gjim83**

### CODE REVIEW

**Comment by gjim83** [comment]
I forgot the `{**dict1, **dict2}` syntax was introduced in 3.5, will fix shortly

**Comment by gjim83** [comment]
I've been working on 3.5 long enough to forget what previous versions are like 😞  this time it's the `if isinstance(members, str)` line biting me in the a$$ during the 2.7 tests

File: napalm_panos/panos.py [hunk 1]
```diff
@@ -534,3 +534,107 @@ def get_interfaces(self):
...
+
+        try:
+            self.device.op(cmd=cmd)
+            interface_info_xml = xmltodict.parse(self.device.xml_root())
+            interface_info_json = json.dumps(
+                interface_info_xml['response']['result']['ifnet']['entry']
+            )
+            interface_info = json.loads(interface_info_json)
+        except KeyError:
+            return ip_interfaces
```
Comment (itdependsnetworks) [inline napalm_panos/panos.py]: not following this one? When would `<show><interface>all</interface></show>` return a condition that meets this?

Comment (gjim83) [reply napalm_panos/panos.py]: good point, I was following the original implementation which had a similar approach, but missed giving that some thought.

Seems the better approach would be remove the `try/except` altogether, and let the call fail if something goes wrong. Come to think of it that should IMO be the same approach for the new `get_interfaces()` too.

**Comment by ktbyers** [comment]
@itdependsnetworks Same applies here...i.e. once you are satisfied with changes, go ahead and merge.

@gjim83 FYI, there is a napalm/base/utils/py23_compat.py for doing PY23 compatibility checks so it could have been something like `isinstance(foo, py23_compat.string_types):`. It is basically a very small subset of `six` library.

Note, no need to change what you have though...

Also going forward, you might want to just format your `expected_result.json` to be more readable (once again don't need to change what you have).

I usually end up doing something like (in the Python shell)

```
>>> import json
>>> a = """{"ethernet1/3.301": {"ipv4": {"172.24.255.1": {"prefix_length": 26}}}, "ethernet1/2": {"ipv4": {"2.2.2.2": {"prefix_length": 24}, "157.191.6.156": {"prefix_length": 24}}}, "ethernet1/1.202": {"ipv4": {"157.191.8.17": {"prefix_length": 28}}}, "ethernet1/4": {"ipv4": {"1.1.1.1": {"prefix_length": 28}, "169.254.0.1": {"prefix_length": 30}}, "ipv6": {"2001::1234": {"prefix_length": 120}, "fe80::d61d:71ff:fed8:fe14": {"prefix_length": 64}}}}
... """
>>> b = json.loads(a)
>>> print(json.dumps(b, indent=4))
{
    "ethernet1/3.301": {
        "ipv4": {
            "172.24.255.1": {
                "prefix_length": 26
            }
        }
    },
    "ethernet1/2": {
        "ipv4": {
            "2.2.2.2": {
                "prefix_length": 24
            },
            "157.191.6.156": {
                "prefix_length": 24
            }
        }
    },
    "ethernet1/1.202": {
        "ipv4": {
            "157.191.8.17": {
                "prefix_length": 28
            }
        }
    },
    "ethernet1/4": {
        "ipv4": {
            "1.1.1.1": {
                "prefix_length": 28
            },
            "169.254.0.1": {
                "prefix_length": 30
            }
        },
        "ipv6": {
            "2001::1234": {
                "prefix_length": 120
            },
            "fe80::d61d:71ff:fed8:fe14": {
                "prefix_length": 64
            }
        }
    }
}
```

**Comment by gjim83** [comment]
@ktbyers, right saw that but I also thought that because the code will work on a `list`, it was clearer to check it's not a `list` as opposed to checking it's a specific type other than `list`.

And thanks for the heads up re: test formatting, will keep in mind.

**Pull request merged**